### PR TITLE
feat: Request body logging in debug level

### PR DESCRIPTION
**Pull request by augustinasrce**
## Description

Please add a description of what your pull request is doing.
 - What was the problem?
>  In order to debug, request body is very helpful.
 - How it is resolved?
>  Introduced request body logging
 - How can we test the change?
> Enabled/disable debug log level and check logs. Unit tests also available.
 - If there are breaking changes, please describe them in detail and why we cannot avoid them.
> None

## Closes issue(s)
 
 #2396

## Checklist
- [x] I have tested this code
- [x] I have added unit test to cover this code
- [x] I have updated the documentation (`README.md` and `/website/docs`)
- [x] I have followed the [contributing guide](CONTRIBUTING.md)

### CODE REVIEW

File: cmd/relayproxy/api/server.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func (s *Server) initRoutes() {
 	s.apiEcho.HidePort = true
 	s.apiEcho.Debug = s.config.IsDebugEnabled()
 	s.apiEcho.Use(custommiddleware.ZapLogger(s.zapLog, s.config))
+	s.apiEcho.Use(custommiddleware.BodyLogger(s.zapLog, s.config))
```
Comment (thomaspoignant) [inline cmd/relayproxy/api/server.go:56]:
```suggestion
	s.apiEcho.Use(middleware.BodyDumpWithConfig(middleware.BodyDumpConfig{
		Skipper: func(c echo.Context) bool {
			return !s.zapLog.Core().Enabled(zap.DebugLevel)
		},
		Handler: func(_ echo.Context, reqBody []byte, _ []byte) {
			log.Debug("Request info", zap.ByteString("request_body", reqBody))
		},
	}))
```

File: cmd/relayproxy/api/middleware/body.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	"go.uber.org/zap"
+)
+
+func BodyLogger(log *zap.Logger, cfg *config.Config) echo.MiddlewareFunc {
+	skipper := func(_ echo.Context) bool { return !cfg.IsDebugEnabled() }
+
+	return middleware.BodyDumpWithConfig(middleware.BodyDumpConfig{
+		Skipper: skipper,
+		Handler: func(_ echo.Context, req []byte, _ []byte) {
+			log.Debug("Request", zap.ByteString("body", req))
```
Comment (thomaspoignant) [inline cmd/relayproxy/api/middleware/body.go:16]: I am not a fan of using `body` as the property name here I would go more with `request_body`, so we can evolve this log in the future.

```suggestion
			log.Debug("Request info", zap.ByteString("request_body", reqBody))
```

File: cmd/relayproxy/api/middleware/body.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	"github.com/thomaspoignant/go-feature-flag/cmd/relayproxy/config"
+	"go.uber.org/zap"
+)
+
+func BodyLogger(log *zap.Logger, cfg *config.Config) echo.MiddlewareFunc {
+	skipper := func(_ echo.Context) bool { return !cfg.IsDebugEnabled() }
+
+	return middleware.BodyDumpWithConfig(middleware.BodyDumpConfig{
+		Skipper: skipper,
+		Handler: func(_ echo.Context, req []byte, _ []byte) {
```
Comment (thomaspoignant) [inline cmd/relayproxy/api/middleware/body.go:15]:
```suggestion
		Handler: func(_ echo.Context, reqBody []byte, _ []byte) {
```
just to be sure this is not misleading and people knows that this does not contains the full request.

**Review by thomaspoignant — COMMENTED** [review]
Hey @augustinasrce thanks for your pull request.

This is a great addition to log the request body, but since we are using an existing middleware I am not sure why we need our own file in the middleware section, I would prefer that we include directly the middleware in the `server.go` file as suggested.

**Comment by augustinasrce** [comment]
> Hey @augustinasrce thanks for your pull request.
> 
> This is a great addition to log the request body, but since we are using an existing middleware I am not sure why we need our own file in the middleware section, I would prefer that we include directly the middleware in the `server.go` file as suggested.

Idea was to keep things simple, without big nested structures (in case it gets bigger).  Since it's quite small at the moment I'll update PR as You suggested.

**Review by thomaspoignant — APPROVED** [review]

**Pull request merged**